packages/control-kripke/src/Control/Kripke.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LiberalTypeSynonyms #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+
+module Control.Kripke where
+
+import Control.Category (Category (..))
+import Data.Kind (Type)
+import Prelude hiding (id, (.))
+
+type All :: (k -> Type) -> Type
+type All f = forall x. f x
+
+infixr 0 -->
+
+type (-->) :: (k -> Type) -> (k -> Type) -> (k -> Type)
+type (-->) f g x = f x -> g x
+
+data World :: Type
+
+type Path :: (World -> World -> Type) -> World -> World -> Type
+data Path step r t where
+  Refl :: Path step r r
+  Step :: Path step r s -> step s t -> Path step r t
```

Comment:
`Step` should be `Trans` if you're using `Refl`.

Reply:
No, this is a normalised reflexive transitive closure, but I'll rename `compose` to `trans`.